ckb/src/main/java/org/nervos/ckb/utils/address/AddressGenerator.java

```diff
@@ -1,36 +1,46 @@
 package org.nervos.ckb.utils.address;
 
+import java.util.Arrays;
+import java.util.List;
 import org.nervos.ckb.address.Network;
 import org.nervos.ckb.type.Script;
 import org.nervos.ckb.utils.Bech32;
 import org.nervos.ckb.utils.Numeric;
 
 /** Copyright © 2019 Nervos Foundation. All rights reserved. */
 public class AddressGenerator extends AddressBaseOperator {
+  private static final List<String> codeHashes =
+      Arrays.asList(
+          SECP_BLAKE160_CODE_HASH,
+          MULTISIG_CODE_HASH,
+          ACP_MAINNET_CODE_HASH,
+          ACP_TESTNET_CODE_HASH);
 
   public static String generate(Network network, Script script) {
-    if (Script.TYPE.equals(script.hashType) && Numeric.cleanHexPrefix(script.args).length() == 40) {
-      if (SECP_BLAKE160_CODE_HASH.equals(Numeric.cleanHexPrefix(script.codeHash))) {
-        // Payload: type(01) | code hash index(00, P2PH) | args
-        String payload = TYPE_SHORT + CODE_HASH_IDX_BLAKE160 + Numeric.cleanHexPrefix(script.args);
-        byte[] data = Numeric.hexStringToByteArray(payload);
-        return Bech32.encode(
-            prefix(network),
-            convertBits(com.google.common.primitives.Bytes.asList(data), 8, 5, true));
-      } else if (MULTISIG_CODE_HASH.equals(Numeric.cleanHexPrefix(script.codeHash))) {
-        // Payload: type(01) | code hash index(01, multi-sig) | args
-        String payload = TYPE_SHORT + CODE_HASH_IDX_MULTISIG + Numeric.cleanHexPrefix(script.args);
-        byte[] data = Numeric.hexStringToByteArray(payload);
-        return Bech32.encode(
-            prefix(network),
-            convertBits(com.google.common.primitives.Bytes.asList(data), 8, 5, true));
-      } else {
-        return generateFullAddress(network, script);
-      }
+    String codeHash = Numeric.cleanHexPrefix(script.codeHash);
+    String args = Numeric.cleanHexPrefix(script.args);
+    if (Script.TYPE.equals(script.hashType)
+        && args.length() == 40
```

Comment:
acp lock's args max length is 22 bytes or 44 characters. @duanyytop

Reply:
I will update later in another pr